### PR TITLE
Support multiple remotes

### LINKED ISSUE #11: Multiple remotes messes things up
**Describe the bug**
I have a repository with multiple remotes (here: `origin` and `production`). Doing a `gh clean-branches --dry-run` I get 

```
Sync branches
fatal: 'origin
production' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Checking out
Your branch is up to date with 'origin/master'.
Pulling
fatal: 'origin
production' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Failed pull, Check for uncomited changes.
```

**To Reproduce**
1. Have a repository with multiple remotes
2. Run `gh clean-branches --dry-run`
3. See error

### PR DESCRIPTION
**Pull request by chrisgoffinet**
fixes #11

### CODE REVIEW

File: gh-clean-branches [hunk 1]
```diff
@@ -36,83 +36,75 @@ end=$(printf '\e[0m')
...
+
+  setopt extended_glob
+  local_branches=("${(f)local_branches_str}")                    # split string by \n to array
+  local_branches=(${local_branches:#* ${default_branch}})        # filter out default_branch
+  local_branches=(${local_branches// ##})                        # trim spaces
+  remote_branches=("${(f)remote_branches_str}")                  # split string by \n to array
+  remote_branches=(${remote_branches:#* ${default_branch}})      # filter out default_branch
+  remote_branches=(${remote_branches// ##})                      # trim spaces
+
+  missing_upstream_branches=(${local_branches:|remote_branches}) # local_branches minus remote_branches
```
Comment (davidraviv) [inline gh-clean-branches:72]: `missing_upstream_branches` will find and delete a local branch that has an upstream in another remote.

Example: 
```bash
⇒  git remote show             
origin-1
origin-2
```

```bash
⇒  git branch -a | cat
* main
  test-branch
  remotes/origin-2/main
  remotes/origin-2/test-branch
  remotes/origin-1/HEAD -> origin/master
  remotes/origin-1/add-module
  remotes/origin-1/main
  remotes/origin-1/master
```
I have two origins, `origin-1` and `origin-2`. 
The local branch `test-branch` has upstream on `origin-2` but doesn't exist on `origin-1`.

When iterating over the upstreams having `upstream_name` as `origin-1`, the branch `test-branch` will be added to `missing_upstream_branches` as it is not included in `remote_branches`. The branch `test-branch` will be deleted in line 79, although it has an upstream in `origin-2`.

**Review by davidraviv — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Please see my comment.

**Comment by chrisgoffinet** [comment]
@davidraviv thanks for the review. I'll work on a fix to handle this case.

**Comment by davidraviv** [comment]
@chrisgoffinet Thanks for the effort in trying to improve the extension.
A [new version](https://github.com/davidraviv/gh-clean-branches/releases/tag/v0.8) is now available that supports multiple upstream.
I hope you can try it out and find it helpful!

**Pull request closed without merging**